### PR TITLE
Enhancement

### PR DESCRIPTION
**Pull request by armin0403**
- added user entity with service and repository
- added UnitOfWork
- added UserViewModel
- added Validator and mapper for User(userviewmodel)
- registered services

### CODE REVIEW

File: NorthwindMVC.Core/Models/Role.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿
+namespace NorthwindMVC.Core.Models
```
Comment (armin0403) [inline NorthwindMVC.Core/Models/Role.cs:2]: Namespace should be  NorthwindMVC.Core

File: NorthwindMVC.Core/Models/UserRole.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿namespace NorthwindMVC.Core.Models
```
Comment (armin0403) [inline NorthwindMVC.Core/Models/UserRole.cs:1]: Namespace should be NorthwindMVC.Core

File: NorthwindMVC.Core/NorthwindMVC.Core.csproj
Comment (armin0403) [inline NorthwindMVC.Core/NorthwindMVC.Core.csproj:1]: Is there a reason for this project to contain Microsoft.EntityFrameworkCore.Tools and Npgsql.EntityFrameworkCore.PostgreSQL packages?

File: NorthwindMVC.Infrastructure/Services/UserService.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+using System.Text;
+using System.Threading.Tasks;
+using NorthwindMVC.Core;
+using NorthwindMVC.Infrastructure.UnitOfWork;
+
+namespace NorthwindMVC.Infrastructure.Services
+{
+    public class UserService : IUserService
+    {
+        private readonly IUnitOfWork UnitOfWork;
```
Comment (armin0403) [inline NorthwindMVC.Infrastructure/Services/UserService.cs:13]: Naming convention when using private variables should be underscore followed by name of the variable

File: NorthwindMVC.Core/Models/Role.cs (hunk 1 above)
Comment (nesherson) [inline NorthwindMVC.Core/Models/Role.cs]: Namespace should be NorthwindMVC.Core

File: NorthwindMVC.Core/Models/UserRole.cs (hunk 2 above)
Comment (nesherson) [inline NorthwindMVC.Core/Models/UserRole.cs]: Namespace should be NorthwindMVC.Core

File: NorthwindMVC.Core/NorthwindMVC.Core.csproj
Comment (nesherson) [inline NorthwindMVC.Core/NorthwindMVC.Core.csproj:1]: Is there a reason for this project to contain Microsoft.EntityFrameworkCore.Tools and Npgsql.EntityFrameworkCore.PostgreSQL packages?

File: NorthwindMVC.Infrastructure/Services/UserService.cs (hunk 3 above)
Comment (nesherson) [inline NorthwindMVC.Infrastructure/Services/UserService.cs:8]: Naming convention when using private variables should be underscore followed by name of the variable

**Pull request merged**